plugins/httpapi/aci.py

```diff
@@ -0,0 +1,191 @@
+# Copyright (c) 2020 Cisco and/or its affiliates.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from __future__ import (absolute_import, division, print_function)
+from ipaddress import ip_address
+__metaclass__ = type
+
+import json
+import re
+import pickle
+import ipaddress
+
+from ansible.module_utils._text import to_text
+from ansible.module_utils.connection import ConnectionError
+from ansible.plugins.httpapi import HttpApiBase
+
+
+class HttpApi(HttpApiBase):
+
+    def __init__(self, *args, **kwargs):
+        super(HttpApi, self).__init__(*args, **kwargs)
+        self.auth = None
+        self.check_auth_from_private_key = None
+        self.backup_hosts = None
+        self.host_counter = 0
+
+    def set_params(self, auth, params):
+        self.params = params
+        self.auth = auth
+
+    def set_backup_hosts(self):
+        try:
+            list_of_hosts = re.sub(r'[[\]]', '', self.connection.get_option("host")).split(",")
+            ipaddress.ip_address(list_of_hosts[0])
+            return list_of_hosts
+        except Exception:
```

Comment:
should we raise more specific exceptions like value error on ip_address function?

Reply:
As of now this is experimental and is not utilized.